include/internal/hpke_util.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_CRYPTO_HPKE_H
+# define OSSL_CRYPTO_HPKE_H
+# pragma once
+
+/* Constants from RFC 9180 Section 7.1 and 7.3 */
+# define OSSL_HPKE_MAX_SECRET 64
+# define OSSL_HPKE_MAX_PUBLIC 133
+# define OSSL_HPKE_MAX_PRIVATE 66
+# define OSSL_HPKE_MAX_NONCE 12
+# define OSSL_HPKE_MAX_KDF_INPUTLEN 64
+
+/*
+ * @brief info about a KEM
+ * Used to store constants from Section 7.1 "Table 2 KEM IDs"
+ * and the bitmask for EC curves described in Section 7.1.3 DeriveKeyPair
+ */
+typedef struct {
+    uint16_t      kem_id; /**< code point for key encipherment method */
```

Comment:
*<

Reply:
see above wrt doxygen, so may disappear in a bit but better resolved on that comment